### PR TITLE
Refactor and add aggregate simulation mode

### PR DESCRIPTION
**Pull request by nathanathan**
Main updates:

1. New flights collection is used instead of legs collection.
2. Geographic criteria for culling flights has been replaced by one that uses cached data.
3. A mode for using aggregate direct flight passenger flows instead of individual flight schedules has been added.
4. A script for comparing the aggregate direct flight mode with the scheduled flight mode has been added.
5. A script for precomputing n-leg airport passenger flows in bulk has been added.

### CODE REVIEW

File: compare_aggregate_flows.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        print "total sets:", count
+        return result
+
+    direct_seat_flows = compute_direct_seat_flows()
+    total_outbound_seats = {
+        airport_id: sum(arrival_seats.values())
+        for airport_id, arrival_seats in direct_seat_flows.items()}
+
+    calculator_with_schedules = AirportFlowCalculator(db)
+    calculator_aggregated_flows = AirportFlowCalculator(db,
```
Comment (toph-allen) [inline compare_aggregate_flows.py]: `AirportFlowCalculator` is the class which actually runs the simulation, right?

Comment (nathanathan) [reply compare_aggregate_flows.py]: Yes

File: simulator/AirportFlowCalculator.py [hunk 2]
```diff
@@ -79,6 +148,28 @@ def __init__(self, db, weight_by_departure_time=True, outgoing_seat_totals=None,
             for leg_num, leg_prob in self.LEG_PROBABILITY_DISTRIBUTION.items()}
         self.max_legs = len(self.LEG_PROBABILITY_DISTRIBUTION) - 1
 
+    def check_logical_layovers(self, origin, destination, layover_set):
```
Comment (toph-allen) [inline simulator/AirportFlowCalculator.py]: I had written a comment, but am going to post what I have written so far in #ibis in Slack.

Comment (toph-allen) [reply simulator/AirportFlowCalculator.py]: Update: My questions are resolved.

**Pull request merged**